core/components/molecules/tabs/Tabs.tsx

```diff
@@ -172,30 +191,52 @@ export const Tabs = (props: TabsProps) => {
       const iconAppearance = activeIndex === index ? 'info' : disabled ? 'disabled' : 'subtle';
       return <Icon data-test="DesignSystem-Tabs--Icon" className="mr-4" name={icon} appearance={iconAppearance} />;
     }
-
     return null;
   };
 
-  const renderTab = (tab: Tab, index: number) => {
-    const { label = '', disabled } = tab as TabConfig;
+  const renderDismissIcon = (tab: Tab, index: number, dismissOnClose: boolean, onDismiss: noop) => {
+    const { disabled, label } = tab as TabConfig;
+    const iconAppearance = activeIndex === index ? 'info' : disabled ? 'disabled' : 'subtle';
+    const tabInfo = { label: label };
+    const onCloseHandler = (e: React.MouseEvent) => {
+      e.stopPropagation();
+      if (dismissOnClose) {
+        const tabData = (tabs as any[]).filter((item) => item !== tabs[index]);
+        setTabs(tabData);
+        if (onDismiss) onDismiss(tabInfo);
+      } else {
+        if (onDismiss) onDismiss(tabInfo);
```

Comment:
`if (onDismiss) onDismiss(tabInfo);` this is similar in both the condition. Please write it outside if-else statement